### PR TITLE
Replace azure-identity AAD references

### PR DESCRIPTION
**Pull request by scottaddie**
Replace Azure AD references in Markdown files and JavaDoc strings with Microsoft Entra ID, per the guidance at https://learn.microsoft.com/azure/active-directory/fundamentals/how-to-rename-azure-ad.

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -274,6 +274,7 @@
     "embedme",
     "encryptor",
     "endtoend",
+    "entra",
```
Comment (joshfree) [inline .vscode/cspell.json:277]: Should this entry be `Entra` and case enforced?

@jairmyree do we have case sensitivity enforced in our cspell configuration - https://cspell.org/docs/case-sensitive/

**Comment by joshfree** [comment]
@scottaddie there are also JavaDoc strings which need to be updated in the source code such as: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java#L75

**Review by joshfree — CHANGES_REQUESTED** [review]
Ideally we should fix the javadoc source comments in this PR too ahead of the November release when they'll be generated and published to microsoft docs

**Comment by scottaddie** [comment]
@joshfree Thanks for the feedback. This is ready for another look.

File: .vscode/cspell.json (hunk 1 above)
Comment (jairmyree) [reply .vscode/cspell.json:277]: @joshfree Our cspell config does not have case-sensitivity enabled. Casing is not enforced.

**Comment by scottaddie** [comment]
@billwert @g2vinay Please review.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredentialBuilder.java [hunk 2]
```diff
@@ -20,11 +20,10 @@
  * for IntelliJ plugin</a> for the IntelliJ IDEA development environment. It enables developers to create, test, and
  * deploy Java applications to the Azure cloud platform. In order to use the plugin authentication as a user or
  * service principal against
- * <a href="https://learn.microsoft.com/azure/active-directory/fundamentals/">Azure Active Directory (Azure AD)
- * </a> is required.
+ * <a href="https://learn.microsoft.com/azure/active-directory/fundamentals/">Microsoft Entra ID</a> is required.
  * The {@link IntelliJCredential} authenticates in a development environment and acquires a token on behalf of the
  * logged-in account in Azure Toolkit for IntelliJ. It uses the logged in user information on the IntelliJ IDE and uses
- * it to authenticate the application against Azure Active Directory.</p>
+ * it to authenticate the application against Microsoft Entra ID.</p>
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredentialBuilder.java:26]: same, should this be, `against Microsoft Entra`

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredentialBuilder.java [hunk 3]
```diff
@@ -39,8 +39,8 @@ public class EnvironmentCredentialBuilder extends CredentialBuilderBase<Environm
     private String authorityHost;
 
     /**
-     * Specifies the Azure Active Directory endpoint to acquire tokens.
-     * @param authorityHost the Azure Active Directory endpoint
+     * Specifies the Microsoft Entra endpoint to acquire tokens.
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredentialBuilder.java:42]: Should this say, `Microsoft Entra ID endpoint` ?

**Review by g2vinay — APPROVED** [review]
In some spots, AAD / Azure AD is replaced with Microsoft Entra ID, and in others it is replaced with Microsoft Entra, 
Is this intentional ? if not, we should fix this, if yes, it LGTM.

File: sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md [hunk 4]
```diff
@@ -129,27 +129,27 @@ az postgres server ad-admin create \
...
 ![postgresql-architecture.png](img/postgresql-architecture.png)
 
-To learn more about using Azure AD with PostgreSQL, see Use (Use Azure Active Directory for authenticating with
-PostgreSQL)[Use Azure Active Directory for authenticating with PostgreSQL]
+To learn more about using Microsoft Entra ID with PostgreSQL, see (Use Microsoft Entra ID for authenticating with
+PostgreSQL)[Use Microsoft Entra ID for authenticating with PostgreSQL].
 
 ### Architecture
 
 This picture shows how the jdbc authentication plugin for PostgreSQl authenticating with managed identity.
```
Comment (billwert) [inline sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md:148]:
```suggestion
This picture shows how the JDBC authentication plugin for PostgreSQL authenticating with managed identity.
```

File: sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md [hunk 5]
```diff
@@ -299,5 +299,5 @@ the [contributing guide](https://github.com/Azure/azure-sdk-for-java/blob/main/C
 [docs]: https://azure.github.io/azure-sdk-for-java/
 [postgresql_product_docs]: https://docs.microsoft.com/azure/postgresql/single-server/overview
```
Comment (billwert) [inline sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md:300]:
```suggestion
[postgresql_product_docs]: https://learn.microsoft.com/azure/postgresql/single-server/overview
```

File: sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md [hunk 6]
```diff
@@ -299,5 +299,5 @@ the [contributing guide](https://github.com/Azure/azure-sdk-for-java/blob/main/C
...
 [quick_start_postgresql]: https://aka.ms/passwordless/quickstart/postgresql
-[Use Azure Active Directory for authenticating with PostgreSQL]:https://docs.microsoft.com/azure/postgresql/single-server/concepts-azure-ad-authentication
+[Use Microsoft Entra ID for authenticating with PostgreSQL]:https://docs.microsoft.com/azure/postgresql/single-server/concepts-azure-ad-authentication
```
Comment (billwert) [inline sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md:302]:
```suggestion
[Use Microsoft Entra ID for authenticating with PostgreSQL]:https://learn.microsoft.com/azure/postgresql/single-server/concepts-azure-ad-authentication
```

File: sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md [hunk 7]
```diff
@@ -200,7 +200,7 @@ Connection connection=DriverManager.getConnection(url,properties);
 
 ### Cloud Configuration
 
-Credentials default to authenticating to the Azure Active Directory endpoint for Azure Public Cloud. To access resources
+Credentials default to authenticating to the Microsoft Entra endpoint for Azure Public Cloud. To access resources
 in other clouds, such as Azure Government or a private cloud, configure credentials with the `azure.authorityHost`
 argument. [AzureAuthorityHosts](https://docs.microsoft.com/java/api/com.azure.identity.azureauthorityhosts?view=azure-java-stable)
```
Comment (billwert) [inline sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md:205]:
```suggestion
argument. [AzureAuthorityHosts](https://learn.microsoft.com/java/api/com.azure.identity.azureauthorityhosts?view=azure-java-stable)
```

File: sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md (hunk 5 above)
Comment (billwert) [reply sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md:300]: it won't let me leave comments on lines that weren't near a modification but there are others here.

File: sdk/identity/azure-identity-extensions/Azure-Database-for-MySQL-README.md [hunk 8]
```diff
@@ -7,9 +7,9 @@
       - [Prepare the working environment](#prepare-the-working-environment)
       - [Create an Azure Database for MySQL server](#create-an-azure-database-for-mysql-server)
       - [Configure a firewall rule for your MySQL server](#configure-a-firewall-rule-for-your-mysql-server)
-      - [Enable Azure AD-based authentication](#enable-azure-ad-based-authentication)
+      - [Enable Microsoft Entra-based authentication](#enable-azure-ad-based-authentication)
```
Comment (billwert) [inline sdk/identity/azure-identity-extensions/Azure-Database-for-MySQL-README.md:10]: Later in the PR you refer to this as `Microsoft Entra ID` - are both acceptable?

File: sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md [hunk 9]
```diff
@@ -114,10 +114,10 @@ az postgres server firewall-rule create \
     --output tsv
 ```
 
-#### Enable Azure AD-based authentication
+#### Enable Microsoft Entra ID-based authentication
```
Comment (billwert) [inline sdk/identity/azure-identity-extensions/Azure-Database-for-PostgreSQL-README.md:117]: For example here - just below you drop `ID`. This is probably fine, just want to make sure.

File: sdk/identity/azure-identity/README.md [hunk 10]
```diff
@@ -1,8 +1,8 @@
 # Azure Identity client library for Java
```
Comment (billwert) [inline sdk/identity/azure-identity/README.md:1]: line 578: `[azuread_doc]: https://learn.microsoft.com/azure/active-directory/` - maybe wanna change the slug.

File: sdk/identity/azure-identity-extensions/Azure-Database-for-MySQL-README.md (hunk 8 above)
Comment (scottaddie) [reply sdk/identity/azure-identity-extensions/Azure-Database-for-MySQL-README.md:10]: Great question. See https://learn.microsoft.com/azure/active-directory/fundamentals/new-name#glossary-of-updated-terminology.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredentialBuilder.java (hunk 3 above)
Comment (scottaddie) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredentialBuilder.java:42]: No. It's a feature/concept, so the ID suffix should be omitted. See https://learn.microsoft.com/azure/active-directory/fundamentals/new-name#glossary-of-updated-terminology.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredential.java [hunk 11]
```diff
@@ -28,11 +28,10 @@
  * for IntelliJ plugin</a> for the IntelliJ IDEA development environment. It
  * enables developers to create, test, and deploy Java applications to the Azure cloud platform. In order to
  * use the plugin authentication as a user or service principal against
- * <a href="https://learn.microsoft.com/azure/active-directory/fundamentals/">Azure Active Directory (Azure AD)
- * </a> is required.
+ * <a href="https://learn.microsoft.com/azure/active-directory/fundamentals/">Microsoft Entra ID</a> is required.
  * The IntelliJCredential authenticates in a development environment and acquires a token on behalf of the
  * logged-in account in Azure Toolkit for IntelliJ. It uses the logged in user information on the IntelliJ IDE and uses
- * it to authenticate the application against Azure Active Directory.</p>
+ * it to authenticate the application against Microsoft Entra ID.</p>
```
[the remark this answers is not in the document]
Comment (scottaddie) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredential.java:34]: No. We're not talking about a specific feature/concept, so we should use the official product name that includes the "ID" suffix.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredentialBuilder.java (hunk 2 above)
Comment (scottaddie) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredentialBuilder.java:26]: See previous comment.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthorizationCodeCredentialBuilder.java [hunk 12]
```diff
@@ -76,10 +76,10 @@ public AuthorizationCodeCredentialBuilder redirectUrl(String redirectUrl) {
     }
 
     /**
-     * Sets the client secret for the authentication. This is required for AAD web apps. Do not set this for AAD native
-     * apps.
+     * Sets the client secret for the authentication. This is required for Microsoft Entra web apps.
+     * Do not set this for Microsoft Entra native apps.
```
Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthorizationCodeCredentialBuilder.java:80]: @g2vinay to confirm but I think this maybe needs a `<p>` to render the line break you want.

**Review by billwert — APPROVED** [review]

**Review by joshfree — APPROVED** [review]
LGTM - thank you @scottaddie

**Pull request merged**